File: sample/src/main/java/com/iktwo/sample/PersonDelegate.java
package com.iktwo.sample;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.iktwo.binder.BindingViewHolder;
import com.iktwo.binder.ViewDelegate;
import com.iktwo.sample.databinding.DelegatePersonBinding;

public class PersonDelegate extends ViewDelegate<BindingViewHolder<DelegatePersonBinding>> {
    private PersonDelegateHandler handler;

    public PersonDelegate(PersonDelegateHandler handler) {
        this.handler = handler;
    }

    @Override
    public int getItemViewType() {
        return 0;
    }

    @Override
    public BindingViewHolder<DelegatePersonBinding> onCreateViewHolder(ViewGroup viewGroup) {
        DelegatePersonBinding binding = DataBindingUtil.inflate(LayoutInflater.from(viewGroup.getContext()),
                R.layout.delegate_person, viewGroup, false);

        binding.setHandler(handler);
        return new BindingViewHolder<>(binding);
    }

    @Override
    public void onBindViewHolder(BindingViewHolder<DelegatePersonBinding> viewHolder, Object item, int position) {
        viewHolder.getBinding().setModel((Person) item);
        viewHolder.getBinding().executePendingBindings();
    }

    @Override
    public boolean canHandleItemTypeAtPosition(@NonNull Object item, int position) {
        if (item instanceof Person) {
            return !((Person) item).isVI;
        }

        return false;
    }
}

File: sample/src/main/java/com/iktwo/sample/PersonDelegate.kt
package com.iktwo.sample

import androidx.databinding.DataBindingUtil
import android.view.LayoutInflater
import android.view.ViewGroup

import com.iktwo.binder.BindingViewHolder
import com.iktwo.binder.ViewDelegate
import com.iktwo.sample.databinding.DelegatePersonBinding

class PersonDelegate(private val handler: PersonDelegateHandler) : ViewDelegate<BindingViewHolder<DelegatePersonBinding>>() {

    override val itemViewType: Int
        get() = 0

    override fun onCreateViewHolder(viewGroup: ViewGroup): BindingViewHolder<DelegatePersonBinding> {
        val binding = DataBindingUtil.inflate<DelegatePersonBinding>(LayoutInflater.from(viewGroup.context),
                R.layout.delegate_person, viewGroup, false)

        binding.handler = handler
        return BindingViewHolder(binding)
    }

    override fun onBindViewHolder(viewHolder: BindingViewHolder<DelegatePersonBinding>, item: Any, position: Int) {
        viewHolder.binding.model = item as Person
        viewHolder.binding.executePendingBindings()
    }

    override fun canHandleItemTypeAtPosition(item: Any, position: Int): Boolean {
        return if (item is Person) {
            !item.isVI
        } else false

    }
}
